backend/src/errors.rs

```diff
@@ -24,12 +24,16 @@ impl fmt::Debug for ParserError {
 
 pub enum TangleError {
     BlockNotFound(String),
+    LanguageNotSupported,
 }
 
 impl fmt::Display for TangleError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
             TangleError::BlockNotFound(msg) => write!(f, "Block tag not found: {}", msg),
+            TangleError::LanguageNotSupported => {
+                write!(f, "Programming language not supported for execution:")
+            }
```

Comment:
Then we can print it like this:
```suggestion
            TangleError::LanguageNotSupported(lang) => {
                write!(f, "Programming language not supported for execution: {}", lang)
            }
```